modules/hooks.nix

```diff
@@ -3331,6 +3375,7 @@ lib.escapeShellArgs (lib.concatMap (ext: [ "--ghc-opt" "-X${ext}" ]) hooks.ormol
             in
             "${hooks.typos.package}/bin/typos ${cmdArgs}";
           types = [ "text" ];
+          excludes = excludesFromConfig;
```

Comment:
`excludes` is expected to be in regex form, whereas `excludesFromConfig` has globs.

Reply:
From my personal testing I can verify that it works when the `typos.toml` file doesn't use globs.

What can we do do mitigate this issue? This issue really needs to be solved for some of our projects.

Reply:
Maybe look for `*`s and `?`s and if they occur, throw an explanation that it's not implemented?
Not fantastic, but leaps better than making users debug or keeping your colleagues/clients waiting.

Ideally, the globs could be translated. https://github.com/hercules-ci/gitignore.nix has inherited an implementation of that from nix-gitignore, but I don't know if it's the same dialect if any. It translates globs into a regex, using a regex... :D ("so now I have three problems"?)

Reply:
Are you suggesting to silently drop excludes? I would hate that as a user and flip a table ;)

Likewise!

Reply:
Agree. Yikes, the whole thing is complicated. I just want `typos` always behave the same, with and without `pre-commit run`.

See https://github.com/cachix/git-hooks.nix/pull/405#issuecomment-2069033631 for a summary of the current discussion.